stout/signal.h

```diff
@@ -35,25 +37,22 @@ inline auto Signal(stout::eventuals::EventLoop& loop, const int signum) {
             using K = std::decay_t<decltype(k)>;
 
             data.k = static_cast<void*>(&k);
-            data.signal.data = &data;
-
-            auto status = uv_signal_init(data.loop, &(data.signal));
-            if (status) {
-              k.Fail(uv_err_name(status));
-            } else {
-              auto error = uv_signal_start_oneshot(
-                  &(data.signal),
-                  [](uv_signal_t* signal, int signum) {
-                    auto& data = *static_cast<Data*>(signal->data);
-                    uv_close((uv_handle_t*) (&data.signal), nullptr);
-                    static_cast<K*>(data.k)->Start(signum);
-                  },
-                  data.signum);
-
-              if (error) {
-                uv_close((uv_handle_t*) (&data.signal), nullptr);
-                k.Fail(uv_err_name(error));
-              }
+
+            uv_handle_set_data(
+                reinterpret_cast<uv_handle_t*>(
+                    data.signal.operator uv_signal_t*()),
```

Comment:
This is... not pleasant to read. 😆 Hopefully renaming that method makes this easier to read too.